### DISCUSSION TITLE
New xDrip Nightly won't install on Android 15 Beta 2.1

### DISCUSSION THREAD

**Discussion by janvitos** [NightscoutFoundation/xDrip#3477] [Bug reports] [answered]
I just tried updating xDrip from 2124051517 to 2124052208 on my Pixel 7 Pro with Android 15 Beta 2.1, but I got the following error message: "App not installed as app isn't compatible with your phone."

It's the first time I see this error. I was able to update to previous xDrip versions while being on Android 15 Beta without issues.

**Comment by Navid200** [comment, marked as answer]
Android 15 beta got an update one week ago causing this.
xDrip is being updated to retire Android 6.
Retiring old versions needs to be done gradually.
Everything will be sorted out before Android 15 is released.

**Reply by janvitos** [reply]
Does that mean current users of Android 15 beta won't be able to update xDrip until the full release of Android 15?

**Reply by Navid200** [reply]
Yes, or until we update xDrip, which is going to happen soon.

**Reply by Navid200** [reply]
I have opened a PR (change proposal) that allows xDrip to be installed on Android 15 beta 2:
https://github.com/NightscoutFoundation/xDrip/pull/3483

**Reply by janvitos** [reply]
> I have opened a PR (change proposal) that allows xDrip to be installed on Android 15 beta 2:
> #3483

Super. Thanks!

**Comment by Navid200** [comment]
This should be resolved now.
Please try the May 29, 2024 release or any after that.
https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.05.29

**Reply by janvitos** [reply]
> This should be resolved now

Was able to install the May 29, 2024 release without issues!

Thank you.

**Comment by parapenT1sta** [comment]
Just my opinion, isn't better to release the https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.05.28 version as the latest Stable version available for Android 6?

**Reply by Navid200** [reply]
The details of what is in every release can be found here: https://navid200.github.io/xDrip/docs/ReleaseNotes.html

Are you suggesting that xDrip would check the Android release on the phone and offer updates based on that?

If you are suggesting that the May 28 release to be promoted to stable release for all phones now, how is that going to help anyone when the next stable release comes out in about 5 months from now?
The May 28 release is not tested enough to be promoted to stable yet anyway.

**Reply by parapenT1sta** [reply]
I was thinking about the people who only install Stable versions because of the update check set by default. The latest version available for them will be the same they are using for months.

**Reply by Navid200** [reply]
OK,
I hear you.
I am not saying you don't have a point.

What I am saying is that even if we make May 28 the stable release, after the next stable release comes out, anyone on Android 6 will get an update that they will not be able to install. 

The differences between our current stable release and the May 28 release are minimal.  They mostly affect G7.  G7 users are supposed to follow the G7 guide, which already tells user which xDrip release they must install.

I even considered creating a parallel repository forked at May 28, 2024 and continuously tracking the main repository minus the Android 6 retirement.
But, that is just not a practical approach.  I am concerned about people who will need to get new phones.
But, considering how outdated Android 6 is today, it may not be a bad idea to update anyway.
Running two repositories in parallel will be a support nightmare.

**Reply by parapenT1sta** [reply]
> OK, I hear you. I am not saying you don't have a point.
 I know.  Don't worry.
 
Even the smartwatches with full android are running Android 7+...

**Reply by Navid200** [reply]
I have no objection to making the May 28, 2024 release our alpha release today and may be in a week or two to our new beta (stable).
But, this requires work on @jamorham part.
I will be more than happy if I am allowed to take over the release promotions.  Or only the alpha release promotions.

**Reply by parapenT1sta** [reply]
> I have no objection to making the May 28, 2024 release our alpha release today and may be in a week or two to our new beta (stable). But, this requires work on @jamorham part. I will be more than happy if I am allowed to take over the release promotions. Or only the alpha release promotions.

I new patch will be released for Libre 2, was great if people with Android 6 could get it. @tzachi-dar already made the patch and it's being tested.

**Reply by Navid200** [reply]
And what do we do when Tzachi releases an update after this one?
What will we do when Stelo comes out in a month or so?
What will we do when the next important G7 update comes out?

Do you have time to help us?
We really don't have a shortage of ideas.  What we do have a shortage of is people who are willing to give us a hand.

**Reply by parapenT1sta** [reply]
You're completly right. Let's forget about Android 6.. it's really old version, released in 2015. 

Sorry for this.

**Reply by Navid200** [reply]
No, it broke my hear when I was not able to use Android 15.  And Android 15 is a train that has already left the station.  It is coming and we have no choice.
I'm not happy about this.   

The problem is that this is just the tip of the iceberg.
We will soon need to retire 7.
